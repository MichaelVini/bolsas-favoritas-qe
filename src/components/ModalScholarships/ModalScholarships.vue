<template>
    <section class="modal">
        <div class="modal-container">
            <div class="modal-dados">
                <button class="close-modal" @click="changeModal">X</button>
                    <h2>Adicionar bolsa</h2>
                    <p>Filtre e adicione as bolsas de seu interesse.</p>
                    <div class="select-city">
                        <span>SELECIONE SUA CIDADE</span>
                        <select name="city" v-model="citySelected" id="city">
                            <option value="" selected>Selecione uma cidade</option>
                            <option 
                                v-for="city in scholarshipsCities"
                                :key="city"  
                                >{{ city }}</option>
                        </select>
                    </div>
                    <div class="select-scholarship">
                        <span>SELECIONE O CURSO DE SUA PREFERÊNCIA</span>
                        <select name="cc" v-model="scholarshipSelected" id="cc">
                            <option value="" selected>Selecione um curso</option>
                            <option 
                            v-for="name in scholarshipsNames" 
                            :key="name" 
                            >{{ name }}</option>
                        </select>
                    </div>
                    <div class="select-modality">
                        <span>COMO VOCÊ QUER ESTUDAR?</span>
                        <div class="modality">
                            <div class="presencial-modality">
                                <input type="checkbox" class="presencial" value="Presencial" v-model="modality" > 
                                <label for="presencial">Presencial</label>
                            </div>
                            <div class="ead-modality">
                                <input type="checkbox" value="EaD" v-model="modality">
                                <label for="ead">A distância</label>
                            </div>
                        </div>
                    </div>
                    <div class="select-max-payment">
                        <span>ATÉ QUANTO PODE PAGAR?</span>
                        <div class="slideContainer">
                            <p>R$ {{ rangeValue }}</p>
                            <input type="range" min="1" max="10000" value="5000" id="myRange" class="slider" v-model="rangeValue">      
                        </div>
                    </div>
                    <div class="filter">
                        <span>Resultado</span>
                        <span>Ordenar por</span>
                    </div>
                    <div v-for="scholarship in scholarshipsFilters"  :key="scholarship"  class="scholarships-list">
                        <input type="checkbox">
                        <img :src="scholarship.university.logo_url" alt="" width="100px">
                        <div class="scholarships-content">
                            <h3>{{ scholarship.course.name }}</h3>
                            <span>{{ scholarship.course.level}}</span>
                            <h4><p> Bolsa de </p> {{ scholarship.discount_percentage}}% </h4>
                            <h4>R$ {{ scholarship.price_with_discount }}/mês</h4>
                        </div>
                    </div>
                    <button>Cancelar</button>
                    <button>Adicionar Bolsas</button>
            </div>
        </div>
    </section>
</template>


<script>
import dataScholarships from '@/services/db.json'

export default {
    data() {
        return{
            rangeValue: "5000",
            scholarships: [],
            scholarshipsCities: [],
            scholarshipsNames: [],
            citySelected: "",
            scholarshipSelected: "",
            modality: []
        }
    },
    props: [
        "changeModal"
    ],
    computed:{
        scholarshipsFilters(){
            const filterByCourseName = this.scholarships.filter((elem) => {
               return elem.course.name === this.scholarshipSelected && elem.price_with_discount <= this.rangeValue 
            })

             const filterByCityName = this.scholarships.filter((elem) => {
                 return elem.campus.city === this.citySelected && elem.price_with_discount <= this.rangeValue 
             })
            
            const filterAll = this.scholarships.filter((elem) => {
                const resul = elem.course.name === this.scholarshipSelected
                 &&  elem.campus.city === this.citySelected && elem.price_with_discount <= this.rangeValue 
                return resul
            }) 

            if(this.citySelected && this.scholarshipSelected == "") {
                return filterByCityName
            } else if(this.scholarshipSelected && this.citySelected == ""){
                return filterByCourseName
            } else {
                return filterAll
            }
        }    
    },
    methods: {
        filterCities(){
            let array = dataScholarships;

            let arrayCities = array.map( response => {
                return response.campus.city
            })

            let arrayCitiesWithoutRepeat = arrayCities.filter((elem, index) => {
                return arrayCities.indexOf(elem) === index;
            })
            // indexOf retorna a primeira posição em que o item do array aparece
            this.scholarshipsCities = arrayCitiesWithoutRepeat;
        },

        filterScholarshipsNames(){
            let array = dataScholarships;

            let arrayScholarshipNames = array.map(response => {
                return response.course.name
            })
            let scholarshipsNameWithoutRepeat = arrayScholarshipNames.filter((elem, index) => {
                return arrayScholarshipNames.indexOf(elem) === index;
            })
            this.scholarshipsNames = scholarshipsNameWithoutRepeat;
        }

    },
    created() {
        this.scholarships = dataScholarships;
        this.filterCities();
        this.filterScholarshipsNames();
    }
}
</script>

<style src="./_modalScholarships.css" scoped>