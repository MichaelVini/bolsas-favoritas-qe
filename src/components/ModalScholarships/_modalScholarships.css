.modal::before {
    content: "";
    position: fixed;
    top: 0px;
    left: 0px;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, .8);
}

.modal {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: absolute;
    top: 0px;
    left: 0px;
    width: 100%;
    padding: 85px 0px 0px 0px;
  }
  
.modal-container {
    position: relative;
    background: white;
    display: grid;
    align-items: end;
    grid-gap: 50px;
    padding: 50px 50px 50px 0;
    max-height: 80vh;
    overflow-y: auto;
}
 
.modal-container select {
    padding: 18px;
    border: solid 2px rgba(202, 202, 202, 0.979);
    background: white;
    border-radius: 6px;
    font-size: 1.1em;
    color: var(--black-text);
}

.modal-container select option{
    padding: 18px;
}

.modal-dados {
    grid-column: 2;
    max-width: 600px;
}

.close-modal {
    position: absolute;
    border: none;
    font-size: 1.3em;
    background-color: rgba(0, 0, 0, 0);
    font-weight: bold;
    top: 5px;
    right: 5px;
    cursor: pointer;
}

.modal-dados p {
    margin: 5px;
}

.modal-dados span {
    font-weight: bold;
    margin-top: 30px;
    margin-bottom: 8px;
}

.select-city {
    display: grid;
}

.select-scholarship {
    display: grid;
}

.select-modality {
    display: grid;
}

.select-modality .modality {
    display: flex;
    margin-bottom: 30px;
}

.select-modality .presencial-modality {
    margin-right: 25px;
}

.slider {
    width: 100%;
    background: var(--blue-primary);
    -webkit-appearance: none;
    outline: none;
    background: var(--blue-primary);
    height: 3px;
    border-radius: 15px;
    margin: 25px 0px;
}

.slider::-webkit-slider-thumb {
    -webkit-appearance: none;
    appearance: none;
    width: 25px;
    height: 25px;
    background: white;
    border-radius: 50%;
    border: solid 2px var(--blue-primary);
    cursor: pointer;
}

.filter {
    display: flex;
    justify-content: space-between;
    margin-bottom: 60px;
}

.scholarships-list {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 8px 0px;
    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
}

.scholarships-list input {
    margin: 0 20px;
}

.scholarships-list .scholarships-content {
    padding: 5px 30px;
    width: 240px;

}

.scholarships-list .scholarships-content h3 {
   margin: 0;
   font-size: 1em;
   padding: 10px 0px;
   color: var(--blue-secondary);
   font-weight: bold;
}
.scholarships-list .scholarships-content h4 {
    display: flex;
    color: rgb(2, 83, 2);
    font-weight: 900;
    font-size: 1.15em;
    margin: 10px 0;
}

.scholarships-list .scholarships-content h4:nth-last-child(2){
    margin-top: 25px;
}

.scholarships-list .scholarships-content span {
    font-weight: 500;
}

.scholarships-list .scholarships-content p {
    margin: 0;
    color: black;
    font-weight: 200;
    margin-right: 5px;
}