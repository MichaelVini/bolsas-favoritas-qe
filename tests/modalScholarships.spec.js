import { createLocalVue, shallowMount } from '@vue/test-utils'
import ModalScholarships from './../src/components/ModalScholarships/ModalScholarships.vue'
import Vuex from 'vuex'
import { it, jest } from '@jest/globals'

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
    state: {
        dataScholarships: [
            {
                "full_price": 2139.64,
                "price_with_discount": 706.08,
                "discount_percentage": 67.0,
                "start_date": "01/08/2019",
                "enrollment_semester": "2019.2",
                "enabled": true,
                "course": {
                  "name": "TESTE",
                  "kind": "Presencial",
                  "level": "Bacharelado",
                  "shift": "Noite"
                },
                "university": {
                  "name": "UNIP",
                  "score": 4.5,
                  "logo_url": "https://www.tryimg.com/u/2019/04/16/unip.png"
                },
                "campus": {
                  "name": "Jardim das Indústrias",
                  "city": "São José dos Campos"
                }
              }
        ]
    },
    mutations: {
        UPDATE_FAVORITE_SCHOLARSHIPS: jest.fn(), 
        UPDATE_SHOWMODAL: jest.fn()
    }
})

describe('ModalScholarships', () => {
    it('should render default state', () => {
        const wrapper = shallowMount(ModalScholarships, {
            localVue,
            store
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
})