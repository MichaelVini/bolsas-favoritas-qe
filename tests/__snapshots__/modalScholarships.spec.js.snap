// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalScholarships should render default state 1`] = `
"<section class=\\"modal\\"><button class=\\"btn-close-modal\\">X</button>
  <div class=\\"modal-container\\">
    <div class=\\"modal-dados\\">
      <h2>Adicionar bolsa</h2>
      <p>Filtre e adicione as bolsas de seu interesse.</p>
      <div class=\\"select-city\\"><span>SELECIONE SUA CIDADE</span> <select name=\\"city\\" id=\\"city\\">
          <option value=\\"\\" selected=\\"selected\\">Selecione uma cidade</option>
          <option>São José dos Campos</option>
        </select></div>
      <div class=\\"select-scholarship\\"><span>SELECIONE O CURSO DE SUA PREFERÊNCIA</span> <select name=\\"cc\\" id=\\"cc\\">
          <option value=\\"\\" selected=\\"selected\\">Selecione um curso</option>
          <option>TESTE</option>
        </select></div>
      <div class=\\"select-modality\\"><span>COMO VOCÊ QUER ESTUDAR?</span>
        <div class=\\"modality\\">
          <div class=\\"presencial-modality\\"><input type=\\"checkbox\\" value=\\"Presencial\\" class=\\"presencial\\"> <label for=\\"presencial\\">Presencial</label></div>
          <div class=\\"ead-modality\\"><input type=\\"checkbox\\" value=\\"EaD\\"> <label for=\\"ead\\">A distância</label></div>
        </div>
      </div>
      <div class=\\"select-max-payment\\"><span>ATÉ QUANTO PODE PAGAR?</span>
        <div class=\\"slideContainer\\">
          <p>R$ 5000</p> <input type=\\"range\\" min=\\"1\\" max=\\"10000\\" value=\\"5000\\" id=\\"myRange\\" class=\\"slider\\">
        </div>
      </div>
      <div class=\\"filter\\"><span>Resultado</span> <span>Ordenar pozzr</span></div>
      <div class=\\"add-favorite-courses\\"><button class=\\"btn-cancel\\">Cancelar</button> <button class=\\"btn-add-courses\\">Adicionar bolsa(s)</button></div>
    </div>
  </div>
</section>"
`;
